CODING_GUIDLINES.md

```diff
@@ -0,0 +1,119 @@
+# Coding Guidelines and Recommendations
+
+The Stack:
+
+- Language: [TypeScript](https://www.typescriptlang.org/)
+- Framework: [React](https://reactjs.org/) + [Knockout](https://knockoutjs.com/)(Deprecated)
+- Bundler: [Webpack](https://webpack.js.org/)
+- Unit Test Runner: [Jest](https://jestjs.io/)
+- Code Formatter: [Prettier](https://prettier.io/)
+- Component Framework: [Fluent](https://developer.microsoft.com/en-us/fluentui#/)
+- Linter: [ESLint](https://eslint.org/) + [TSLint](https://palantir.github.io/tslint/)(Deprecated)
+- End to End Test Runner: [Jest](https://jestjs.io/)
+- End to End Test Framework: [Puppeteer](https://developers.google.com/web/tools/puppeteer)
+- Transpiler: [Babel](https://babeljs.io/) + [TypeScript](https://www.typescriptlang.org/)
+
+## Requirements
+
+Cosmos Explorer has been under constant development for over 5 years. As a result there are many different patterns and practices in the code base. This document serves as a guide to how we write code and helps avoid propagating practices which are no longer preferred. Each requirement in this document is labelled and color-coded to show the relative importance. In order from highest importance to lowest importance:
+
+✅ DO this. If you feel you need an exception, engage with the project owners _prior_ to implementation.
+
+⛔️ DO NOT do this. If you feel you need an exception, engage with the project owners _prior_ to implementation.
+
+☑️ YOU SHOULD strongly consider this but it is not a requirement. If not following this advice, please comment code with why and proactively begin a discussion as part of the PR process.
+
+⚠️ YOU SHOULD NOT strongly consider not doing this. If not following this advice, please comment code with why and proactively begin a discussion as part of the PR process.
+
+💭 YOU MAY consider this advice if appropriate to your situation. Other team members may comment on this as part of PR review, but there is no need to be proactive.
+
+## Development Environment
+
+☑️ YOU SHOULD use VSCode and install the following extensions:
+
+- [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
+- [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
+
+This setup will catch most linting/formatting/type errors as you develop.
+
+☑️ YOU SHOULD use the [GitHub CLI](https://cli.github.com/). It has helpful workflows for submitting PRs as well as for checking out other team members PRs.
+
+💭 YOU MAY use Windows, Linux (including WSL), or OSX. We have team members on all three environments.
+
+✅ DO maintain cross platform compatability when modifying any engineering systems.
+
+## Formatting
+
+✅ DO use [Prettier](https://prettier.io/) to format your code. This will be preformed automatically if using the recommended editor setup. Otherwise `npm run format` will also format code.
```

Comment:
>DO use [Prettier [](start = 2, length = 16)

how about we have this close to where you talk about prettier earlier in line 34. that will give more context.

Reply:
This is a good point. Maybe I can move everything in the "Stack" section down along side the todos